app/scripts/controllers/detect-tokens.js

```diff
@@ -0,0 +1,93 @@
+const contracts = require('eth-contract-metadata')
+const {
+    MAINNET,
+    } = require('./network/enums')
+
+// By default, poll every 3 minutes
+const DEFAULT_INTERVAL = 180 * 1000
+
+/**
+ * A controller that polls for token exchange
+ * rates based on a user's current token list
+ */
+class DetectTokensController {
+  /**
+   * Creates a DetectTokensController
+   *
+   * @param {Object} [config] - Options to configure controller
+   */
+  constructor ({ interval = DEFAULT_INTERVAL, preferences, network } = {}) {
+    this.preferences = preferences
+    this.interval = interval
+    this.network = network
+    this.contracts = contracts
+  }
+
+   /**
+   * For each token in eth-contract-metada, find check selectedAddress balance.
+   *
+   */
+  async exploreNewTokens () {
+    if (!this.isActive) { return }
+    if (this._network.getState().provider.type !== MAINNET) { return }
+    let detectedTokenAddress, token
+    for (const address in this.contracts) {
+        const contract = this.contracts[address]
+        if (contract.erc20 && !(address in this.tokens)) {
+          detectedTokenAddress = await this.fetchContractAccountBalance(address) 
+          if (detectedTokenAddress) {
+            token = this.contracts[detectedTokenAddress]
+            this._preferences.addToken(detectedTokenAddress, token['symbol'], token['decimals'])
+          }
+        }
+        // etherscan restriction, 5 request/second, lazy scan
+        setTimeout(() => {}, 200)
+    }
+  }
+
+   /**
+   * Find if selectedAddress has tokens with contract in contractAddress.
+   *
+   * @param {string} contractAddress Hex address of the token contract to explore.
+   * @returns {string} Contract address to be added to tokens.
+   *
+   */
+  async fetchContractAccountBalance (contractAddress) {
+    const address = this._preferences.store.getState().selectedAddress
+    const response = await fetch(`https://api.etherscan.io/api?module=account&action=tokenbalance&contractaddress=${contractAddress}&address=${address}&tag=latest&apikey=NCKS6GTY41KPHWRJB62ES1MDNRBIT174PV`)
+    const parsedResponse = await response.json()
+    if (parsedResponse.result !== '0') {
```

Comment:
the theory is good, if `result > 0` you have tokens
but for [real case]( 
https://api.etherscan.io/api?module=account&action=tokenbalance&contractaddress=0x7E43581b19ab509BCF9397a2eFd1ab10233f27dE&address=0x87efa7f59bAA8e475F181B36f77A3028494a2cf6&tag=latest&apikey=NCKS6GTY41KPHWRJB62ES1MDNRBIT174PV)
```json
{"status":"1","message":"OK","result":"8121799999999999999558"}
```
i have 8121 tokens, because contract [code says](https://etherscan.io/address/0x7E43581b19ab509BCF9397a2eFd1ab10233f27dE#code)
`uint8 public decimals = 18;`
so let say i have 
```json
{"status":"1","message":"OK","result":"1000"}
```
1000 > 0 but it still means nothing because 1000 * 1e-18 = 1e-15 = 0